XamlAnimatedGif/Animator.cs

```diff
@@ -460,6 +519,19 @@ private async Task<Stream> GetIndexStreamAsync(GifFrame frame, CancellationToken
             return lzwStream;
         }
 
+        private async Task<byte[]> GetIndexBytesAsync(int frameIndex, CancellationToken cancellationToken)
+        {
+            var startPosition = _metadata.Frames[frameIndex].ImageData.CompressedDataStartOffset;
+            var endPosition = _metadata.Frames.Count == frameIndex + 1 ? _sourceStream.Length : _metadata.Frames[frameIndex + 1].ImageData.CompressedDataStartOffset - 1;
+
+            cancellationToken.ThrowIfCancellationRequested();
+            _sourceStream.Seek(startPosition, SeekOrigin.Begin);
+            byte[] result = new byte[endPosition - startPosition];
```

Comment:
This allocates a new buffer for every frame, which puts pressure on the GC. It would be better to allocate a single buffer large enough for the largest frame (easy to calculate from the frames metadata), and pass it to that method. It means you will get a buffer larger than necessary, and pass that to `LzwDecompressStream`, but IIRC it will just ignore the extra bytes.

Reply:
very great idea. Now I allocate a big enough byte[] to be used here